.github/actions/setup/action.yml

```diff
@@ -67,35 +67,40 @@ runs:
         version: ${{ env.SOLANA_VERSION }}
         cache: true
 
+    - name: Cache Cargo Components
+      id: cache-cargo
+      if: ${{ inputs.cargo-cache-key }}
+      uses: actions/cache@v4
+      with:
+        path: |
+          ~/.cargo/bin/
+          ~/.cargo/registry/index/
+          ~/.cargo/registry/cache/
+          ~/.cargo/git/db/
+          target/
+        key: ${{ runner.os }}-${{ inputs.cargo-cache-key }}-${{ hashFiles('**/Cargo.lock') }}
+
     - name: Install 'cargo-audit'
-      if: ${{ contains(inputs.components, 'audit') }}
+      if: ${{ steps.cache-cargo.outputs.cache-hit != 'true' && contains(inputs.components, 'audit') }}
```

Comment:
Rather than adding all of these checks, which might could have a false positive since the extra tools are not part of the lockfile, it's also possible to use https://github.com/taiki-e/cache-cargo-install-action to install tools